service/history/workflowTaskHandlerCallbacks.go

```diff
@@ -411,7 +412,8 @@ func (handler *workflowTaskHandlerCallbacksImpl) handleWorkflowTaskCompleted(
 
 	currentWorkflowTask := ms.GetWorkflowTaskByID(token.GetScheduledEventId())
 	if !ms.IsWorkflowExecutionRunning() || currentWorkflowTask == nil || currentWorkflowTask.Attempt != token.Attempt ||
-		currentWorkflowTask.StartedEventID == common.EmptyEventID {
+		currentWorkflowTask.StartedEventID == common.EmptyEventID ||
+		(token.StartedEventId != common.EmptyEventID && token.StartedEventId != currentWorkflowTask.StartedEventID) {
```

Comment:
If you want to see this working on bench-go then you can take https://github.com/mmcshane/temporal/blob/ec1ee80acf5fb6ed961b59dd741fd4b2001c2ad5/service/history/workflowTaskHandlerCallbacks.go#L1030-L1048. But not blocking.